src/esm_runscripts/filedicts.py

```diff
@@ -359,7 +436,67 @@ def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
         else:
             raise Exception(f"Cannot identify the path's type of {path}")
 
-    def _check_file_syntax(self):
+    @classmethod
+    def wild_card_check(self, source_pattern: list, target_pattern: list) -> True:
```

Comment:
Class methods should use `cls` instead of `self` as the first argument. The actual argument name does not matter, so it is just for clarity. Here is an example:

For regular methods, you get an instance of the object, e.g.:
```python
class MyThing:
    def do_stuff(self):
        print(f"Hooray, I am {self}")

    @classmethod
    def do_class_stuff(cls):
        print(f"Woohoo, I am {cls}")

a = MyThing
a.do_stuff()
MyThing.do_class_stuff()
a.do_class_stuff()
```

Example output:
```
Hooray, I am <__main__.MyThing object at 0x7fa636dd7fa0>
Woohoo, I am <class '__main__.MyThing'>
Woohoo, I am <class '__main__.MyThing'>
```

And in fact, given that you never use `self` at all, this can just be a `staticmethod`

```suggestion
    @staticmethod 
    def wild_card_check(source_pattern: list, target_pattern: list) -> True:
```

Reply:
I see! This clarifies a lot, thanks! I'll copy the link to this comment to my technical notes for the time I forget ^^